### PR TITLE
Feat/ccn3 1754/your children task

### PR DESCRIPTION
**Pull request by cruddasj**
## Description

Addition of workflow logic to support completion of the "Your Children" task.

## Todo's before PR

- [X] Branch has been rebased against develop
- [X] Unit tests passed (`make test`)
- [X] PR named appropriately - see [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Commits-guideline)
- [X] PR description describes the overall goals of PR commits
- [X] Templates have been checked against the relevant user-story

## Migrations

- [X] Null fields in models specifies default value
- [X] You generated and applied migrations (`make migrate`)
- [X] You updated fixtures (`make export`)

## PR Todo's

Please refer to PR [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-submit-a-pull-request)

- [ ] Specified reviewers (even if it's yourself)
- [ ] Specified assignees (those who'll merge)
- [ ] Specified labels

## PR review

Please refer to PR review [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-review-a-pull_request)

- [ ] Code syntax formatting checked
- [ ] Debug messages are absent

## PR merge

Select only one:

- [ ] Should be merged?
- [X] Should be rebased?
- [ ] Should be squashed?

Please refer to PR merge [guide](https://github.com/InformedSolutions/OFS-MORE-DevOps-Tooling/wiki/Pull-requests#how-to-merge-a-pull_request)

### CODE REVIEW

File: application/tests/tests_application/test_your_children.py [hunk 1]
```diff
@@ -0,0 +1,618 @@
+"""
+Tests targetting the payment process
```
Comment (ricaduchateau) [inline application/tests/tests_application/test_your_children.py]: Comment needs updating

**Comment by ricaduchateau** [comment]
Hi @cruddasj, most of my comments are about comments not being up to date, but these are not urgent. I have two questions about table names and one about quote marks in a ValidationError.

File: application/models/child.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        if len(self.middle_names) > 0:
+            concatenated_name = self.first_name + " " \
+                                + self.middle_names + " " + self.last_name
+        else:
+            concatenated_name = self.first_name + " " + self.last_name
+
+        return concatenated_name
+
+    class Meta:
+        db_table = 'CHILD'
```
[the remark this answers is not in the document]
Comment (cruddasj) [reply application/models/child.py]: This originally was named CHILD_OUTSIDE_HOME but the workflow logic means they could be inside or outside the home...

File: application/forms/your_children.py [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+    def clean_children_living_with_childminder_selection(self):
+        """
+        Validator method to ensure options selected are permissible.
+        If for example, names have been chosen as well as 'none' an error should be raised
+        """
+        selections = self.cleaned_data['children_living_with_childminder_selection']
+
+        if len(selections) > 1 and 'none' in selections:
+            raise forms.ValidationError('Please select your children''s names or none.')
```
[the remark this answers is not in the document]
Comment (cruddasj) [reply application/forms/your_children.py]: @ricaduchateau - a double single quote escapes the single quote so this should be okay as-is

**Pull request merged**